### DISCUSSION TITLE
Crossplane with Multiple ProviderCredentials and Azure User Managed Identities

### DISCUSSION THREAD

**Discussion by avosepp** [crossplane/crossplane#5430] [Q&A]
Is there someone here with experience with Crossplane + User Managed Identities in Azure? I need some knowledge.
1. As I understand it, a Workload Identity is assigned to a pod with a token that is loaded into the pod’s filesystem. Crossplane can have more than one ProviderCredential. But if I can only load one token into Crossplane, how can I have multiple ProviderCredentials if they use Workload Identities? Am I limited to a single User Managed Identity if I go that route? Can someone walk me through this? Thank you. :pray:

**Comment by evgheni7** [comment]
https://docs.upbound.io/providers/provider-azure/authentication/#user-assigned-managed-identity

**Comment by madi-parloa** [comment]
From what I understand it is only possible to have a single UAMI with Crossplane, which is very bad security wise. Means we cannot limit permissions scope. If I am wrong please let me know. It would be great to have that fixed.

**Reply by ManuInNZ** [reply]
After chasing a few issues, I still can't find a good doc to explain how to do that and if possible with namespaced resources and providerconfig    

Any joy from someone else?

**Comment by mdlopresti** [comment]
Found this while I was trying to find the answer to this.  Wanted to post a link back here to help others in the future.

https://github.com/crossplane-contrib/provider-azure/issues/329#issuecomment-2595820451

**Reply by ManuInNZ** [reply]
Using it, I have something working but I am not sure if it is the best.

In short:
- in your `crossplane-system` namespace, you have the providers all referring to the `DeploymentRuntimeConfig` with a given `serviceAccountName` and obviously the matching SA created in that namespace, e.g. `azure-family`. 
- In your namespaces, you have `ProviderConfig` with the `OIDCTokenFile` source details and then all compositions/resources of the namespace referring to the `ProviderConfig`
- in the respective UAMI resources, have federated creds using the SA in the `crossplane-system` ( subject identifier: `system:serviceaccount:crossplane-system:azure-family`)

Doing so, while invoking the provider, it will override the creds used by the provider and apparently needs the sa in crossplane-system for the Issuer validation. 

Hope that makes sense
